src/components/Flyout/FlyoutWindow.part.tsx

```diff
@@ -348,17 +354,38 @@ export class FlyoutWindowInt extends React.Component<FlyoutWindowProps, FlyoutWi
   private setFlyoutRef = (el: HTMLDivElement) => {
     if (el) {
```

Comment:
Potential leak (maybe also apparent somewhere else in the library) - if the callback is used multiple times (either for the same DOM node or - even worse - for multiple nodes) we connect multiple handlers and only disconnect the last one that has been set.

Idea: Always before appending call a function that just executes

```js
    if (this.flyoutContainer) {
      this.flyoutContainer.removeEventListener('scroll', this.onScroll);
    }
```

Reply:
I moved it to componentDidMount